src/clients/scc_auto.rb

```diff
@@ -193,16 +193,28 @@ def write
       return false unless set_registration_url
 
       # update the registration in AutoUpgrade mode if the old system was registered
-      if Mode.update && old_system_registered?
-        # drop all obsolete repositories and services (manual upgrade contains a dialog
-        # where the old repositories are deleted, in AY we need to do it automatically here)
-        # Note: the Update module creates automatically a backup which is restored
-        # when upgrade is aborted or crashes.
-        repo_cleanup
-
-        ret = ::Registration::UI::OfflineMigrationWorkflow.new.main
-        log.info "Migration result: #{ret}"
-        return ret == :next
+      if Mode.update
+        if old_system_registered?
+          # drop all obsolete repositories and services (manual upgrade contains a dialog
+          # where the old repositories are deleted, in AY we need to do it automatically here)
+          # Note: the Update module creates automatically a backup which is restored
+          # when upgrade is aborted or crashes.
+          repo_cleanup
+
+          ret = ::Registration::UI::OfflineMigrationWorkflow.new.main
+          log.info "Migration result: #{ret}"
+          return ret == :next
+        else
+          # profile wants to do registration, but old system is not registered. So complain.
```

Comment:
Maybe I'd add the `TRANSLATOSRS:` mark and move the comment to the translated text as this context information might be useful for the translators.